### PR TITLE
Clean up and improve documentation

### PR DESCRIPTION
**Pull request by ahcorde**
Clean up and improve documentation

@brawner

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Review by brawner — APPROVED** [review]

**Comment by ahcorde** [comment]
I removed two methods. Running CI in for all the pacakges to check that these symbols are not used. And running test only for class_loader

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10865/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6230/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8851/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10756/)

**Comment by brawner** [comment]
Is there a reason you're removing the methods? I thought code was under an API freeze?

**Comment by ahcorde** [comment]
I forgot to removed these headers when I changed Poco for rcpputils::SharedLibrary. There is no implementation for these methods. Should we keep it?

**Review by brawner — APPROVED** [review]
I think this is fine then, but could you get approval from a more seasoned maintainer? I'm not sure what the policy is.

**Comment by ahcorde** [comment]
@wjwwood can you have a look ?

File: include/class_loader/class_loader.hpp
Comment (wjwwood) [inline include/class_loader/class_loader.hpp:65]: `systemLibrarySuffix()` no longer exists right?

File: include/class_loader/class_loader.hpp
Comment (wjwwood) [inline include/class_loader/class_loader.hpp]:
```suggestion
   * @brief Indicates if a library is loaded within the scope of this ClassLoader.
   *
   * Note that the
```

Comment (wjwwood) [reply include/class_loader/class_loader.hpp]: (check the other lines for extra spaces)

**Review by wjwwood — APPROVED** [review]
lgtm, some small comments.

File: include/class_loader/class_loader.hpp
Comment (ahcorde) [reply include/class_loader/class_loader.hpp:65]: Yes, I forgot to remove it

File: include/class_loader/class_loader.hpp
Comment (ahcorde) [reply include/class_loader/class_loader.hpp]: Added some extra lines: https://github.com/ros/class_loader/pull/156/commits/3239f3e30a621a4e3c3c3d909c11edfdb2f79084

**Review by wjwwood — APPROVED** [review]

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10936/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6282/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8905/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10838/)

**Pull request merged**
